src/bbs2gh/Handlers/MigrateRepoCommandHandler.cs

```diff
@@ -215,7 +215,18 @@ private async Task ImportArchive(MigrateRepoCommandArgs args, string archiveUrl
 
         args.GithubPat ??= _environmentVariableProvider.TargetGithubPersonalAccessToken();
         var githubOrgId = await _githubApi.GetOrganizationId(args.GithubOrg);
-        var migrationSourceId = await _githubApi.CreateBbsMigrationSource(githubOrgId);
+
+        string migrationSourceId;
+
+        try
+        {
+            migrationSourceId = await _githubApi.CreateBbsMigrationSource(githubOrgId);
+        }
+        catch (OctoshiftCliException ex) when (ex.Message.Contains("not have the correct permissions to execute"))
+        {
+            var exception = MigrateRepoErrorHandler.DecorateInsufficientPermissionsException(ex, args.GithubOrg);
+            throw exception;
+        }
```

Comment:
To rewrite this with my suggested `MessasgeEnhancer` class:
```suggestion
        catch (OctoshiftCliException ex) when (ex.Message.HasEnhancement())
        {
            var message = ex.Message.Enhance((original, improved) => $"{original}. {string.Format(improved, args.GithubOrg)}" ;
            throw new OctoshiftCliException(message, ex);
        }
```

Note that in this particular case, we need to use the factory because the improved message is actually a template that should contain the org name. For this kind of more dynamic creation we cannot use string interpolation but we can use `string.Fromat` instead.